### ISSUE TITLE
Porting to Electron?

### ISSUE THREAD

**Issue by greggman** [mimecuvalo/fireftp#195]
Fireftp is awesome! Thanks so much. I've used it since it shipped many years ago

If I understand correctly though it will very soon no longer run on Firefox as they remove support for old plugins. Also, Apps style pages (like FireFTP) will not be supported even in the new extension system.

Any thoughts about porting to maybe Electron (or the Firefox equivalent?) 

I'm guessing it would be not a small amount of work given Fireftp is based on XUL and would need to be ported HTML (or maybe it's not that far off).

Anyway, of course it would be a major undertaking. But it would be super nice to see fireftp live on.

**Comment by mimecuvalo** [comment]
Development on FireFTP is slowing down since Firefox is dropping support for it in version 57 later in the year.  Hence, motivation for working on it isn't the highest currently. I'll probably be porting FireFTP at least to PaleMoon so that the extension survives.

Electron would be cool but would need a lot of work and I have a full-time job to attend to already.  Patches welcome :)

**Comment by tdulcet** [comment]
> Also, Apps style pages (like FireFTP) will not be supported even in the new extension system.

Not exactly. You would just switch from using `chrome://` to `moz-extension://`, etc. For example, see Mozilla's Lightbeam add on which they ported from XUL: https://github.com/mozilla/lightbeam to a Web Extension: https://github.com/mozilla/lightbeam-we.
